week11/longest_increasing_subsequence.cpp

```diff
@@ -0,0 +1,20 @@
+// TC: O(n^2)
```

Comment:
For some interviews, `O(N^2)` is too much. Try to do it in `O(NlogN)`.

Reference: https://leetcode.com/problems/longest-increasing-subsequence/discuss/1326308/C%2B%2BPython-DP-Binary-Search-BIT-Segment-Tree-Solutions-Picture-explain-O(NlogN)